pkg/files/files_test.go

```diff
@@ -169,4 +170,42 @@ var _ = Describe("should handle files properly", func() {
 
 		Expect(strings.Replace(buffer.String(), "\r", "", -1)).To(BeEquivalentTo("test\n")) // remove \r from windows
 	})
+
+	_ = It("should overwrite permission sets on files", func() {
+		pathToTestDir := "../../assets/tests/issue-51/file"
+		pathToTestFile := filepath.Join(pathToTestDir, "permission.txt")
+
+		os.Chmod(pathToTestFile, 0777)
+		Expect(GetFilePermission(pathToTestFile)).To(BeEquivalentTo(0777))
+
+		Expect(LoadFromDisk(root, pathToTestDir)).To(BeNil())
+		file := root.File(paths.Of("permission.txt"))
+		Expect(file).To(Not(BeNil()))
+
+		file.WithPermission(os.FileMode(0701))
+		Expect(WriteToDisk(root, pathToTestDir, true)).To(BeNil())
+		Expect(GetFilePermission(pathToTestFile)).To(BeEquivalentTo(os.FileMode(0701)))
+	})
+
+	_ = It("should overwrite permission sets on folders", func() {
```

Comment:
Just one tiny suggestion here: `folder` is the term that Microsoft popularised with Windows. If possible, I always try to use the older term `directory`.

Reply:
I'll change it 👍